libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Trace/TraceMonad.hs

```diff
@@ -444,7 +444,9 @@ mockChainProp proof n genTxAndUpdateEnv propf = do
 --   conjoin [p pstate0 sig1 state1, p state0 sig2 state2, p state0 sig3 state3]
 -- test that 'p' holds between state0 and stateN for all N
 stepProp :: (MockChainState era -> MockBlock era -> MockChainState era -> Property) -> (Trace (MOCKCHAIN era) -> Property)
-stepProp p tr = conjoin (map (\(sig, st) -> p state0 sig st) (tail (zip sigs states)))
+stepProp p tr = case (zip sigs states) of
+  [] -> property True
+  (_ : more) -> conjoin (map (\(sig, st) -> p state0 sig st) more)
```

Comment:
Besides the stylistic suggestion I have a question: Is there a guarantee that there will be as many signals as there are states? Otherwise `zip`/`zipWith` will discard the extras, which is ok if it expected.


```suggestion
stepProp p tr = case zipWith (p state0) sigs states of
  [] -> property True
  (_ : more) -> conjoin more
```

Reply:
There is always one more state than signals. But the number of these depends on lots of factors.